src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11105,7 +11109,7 @@ private LambdaExpressionSyntax ParseLambdaExpression(SyntaxToken asyncToken)
                     arrow = CheckFeatureAvailability(arrow, MessageID.IDS_FeatureLambda);
                     exclamation = null;
                 }
-                exclamation = CheckFeatureAvailability(exclamation, MessageID.IDS_ParameterNullChecking);
+                exclamation = !(exclamation is null) ? CheckFeatureAvailability(exclamation, MessageID.IDS_ParameterNullChecking) : null; ;
```

Comment:
consider inverting conditional expressions so the parentheses aren't needed, I really don't have a strong preference though